kubernetes_env/benchmark/benchmark.py

```diff
@@ -20,115 +24,137 @@
+    timestamp = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(time.time()))
+    plot_name = "-".join(filters) + timestamp
+    if fortio:
+        for df in dfs:
+            sns.lineplot(data=df, x="Latency (ms)", y="Percent")
+        plt.legend(labels=filters)
+        plt.title(f"Fortio: {title}")
+    else:
+        plot = sns.displot(dfs, kind="ecdf")
+        plot.set(title=plot_name)
     util.check_dir(GRAPHS_DIR)
     plt.savefig(f"{GRAPHS_DIR}/{plot_name}.png")
     log.info("Finished plotting. Check out the graphs directory!")
     return util.EXIT_SUCCESS
 
 
-def run_fortio(platform, threads, qps, run_time, file_name):
+def run_fortio(url, platform, threads, qps, run_time, file_name):
     util.check_dir(DATA_DIR)
-    _, _, gateway_url = kube_env.get_gateway_info(platform)
     output_file = str(DATA_DIR.joinpath(f"{file_name}.json"))
     fortio_dir = str(FORTIO_DIR)
     cmd = f"{fortio_dir} "
     cmd += f"load -c {threads} -qps {qps} -jitter -t {run_time}s -json {output_file} "
-    cmd += f"http://{gateway_url}/productpage"
+    cmd += f"{url}"
     with open(output_file, "w") as f:
         fortio_res = util.exec_process(cmd,
                                        stdout=subprocess.PIPE,
                                        stderr=subprocess.PIPE)
         return fortio_res
 
 
-def start_benchmark(fortio, filter_dirs, platform, threads, qps, time):
+def burst_loop(url, threads, qps, run_time, queue):
+    def send_request(_):
+        try:
+            # What should timeout be?
+            res = requests.get(url)
+            ms = res.elapsed.total_seconds() * 1000
+            return ms
+        except requests.exceptions.ReadTimeout:
+            pass
+
+    log.info("Starting burst...")
+
+    with ThreadPoolExecutor(max_workers=threads) as p:
+        current = datetime.now()
+        end = current + timedelta(seconds=run_time)
+        while current < end:
+            results = list(
+                p.map(send_request, range(qps), chunksize=qps // threads))
+            current += timedelta(seconds=1)
+            queue.put(results)
+
+
+def do_burst(url, platform, threads, qps, run_time):
+    queue = Queue()
+    _, _, gateway_url = kube_env.get_gateway_info(platform)
+    p = Process(target=burst_loop, args=(
+        url,
+        threads,
+        qps,
+        run_time,
+        queue,
+    ))
+    p.start()
+    p.join()
+    output = []
+    qsize = queue.qsize()
+    for i in range(qsize):
+        output += queue.get()
+    return output
+
+
+def start_benchmark(custom, filter_dirs, platform, threads, qps, run_time):
     if kube_env.check_kubernetes_status() != util.EXIT_SUCCESS:
         log.error("Kubernetes is not set up."
                   " Did you run the deployment script?")
         return util.EXIT_FAILURE
 
+    _, _, gateway_url = kube_env.get_gateway_info(platform)
+    product_url = f"http://{gateway_url}/productpage"
+    log.info(f"Gateway URL: {product_url}")
```

Comment:
https://pycodequ.al/docs/pylint-messages/w1203-logging-fstring-interpolation.html

When you have lazy logging like this, it is recommended to use % arguments instead (because otherwise you might evaluate the string even if you do not need to) I'd recommend pylint for small stuff like this.